Mostly add syntax highlighting ability

ComFreek opened the pull request:
The syntax highlighter works with Pygments. By chance, I already have an MMT Pygments plugin: https://github.com/ComFreek/mmt-pygments-lexer

So in the future we could post MMT code into the documentation *and* get it syntax-highlighted. As of now, it already pays off for the various Scala snippets in there.

You can see it live as of now here: https://comfreek.github.io/uniformal.github.io/doc/tutorials/applications/getting-started.html

![image](https://user-images.githubusercontent.com/1827709/64269689-9ec0d080-cf3a-11e9-84bf-91e1e92a5185.png)

ComFreek committed:
Fix YAML syntax error

ComFreek committed:
docs: escape Markdown links

ComFreek committed:
Add syntax highlighting

ComFreek committed:
Fix HTML syntax error

ComFreek committed:
Make asset paths relative to deployed URI

This allows forked repos' gh-pages' to render well and locally as well.

ComFreek committed:
Fix erratic massreplacement of /doc in text

ComFreek commented:
Getting an e-mail notification from GitHub saying 'pygments' is actually not supported and silently replaced by 'rouge'. Damn, so I guess, at least we've got Scala highlighting now, but MMT highlighting will probably be impossible. (Unless deployment is done via Travis, which doesn't restrict Pygments!)

tkw1536 commented on _data/menu.yml:
```diff
@@ -56,8 +56,8 @@
       url: '/doc/philosophy/intros.html'
-    - title: 'The OMDoc Format'
-      url: '/doc/philosophy/omdoc.html' 
+    - title: 'The OM/doc Format'
```
Not sure I like the OM/doc spelling. The name is OMDOC (without any /s)

kohlhase replied:
Actually, the name is `OMDoc`.

ComFreek replied:
Ah, that was an unintentional mistake of mass regex replacement of `/doc` URLs (relativizing them). Reverted in latest commit.

tkw1536 reviewed (changes requested):
Apart from the one spelling problem LGTM

tkw1536 commented:
Pity about pigments not being supported. We could always consider using Travis and use github-pages as deploy only.

ComFreek committed:
Switch to Travis for building

previously we relied on GitHub for Jekyll building, but
that does not support Pygments highlighting

ComFreek committed:
Merge branch 'master' of github.com:ComFreek/uniformal.github.io

ComFreek committed:
Fix Travis

ComFreek committed:
Fix Travis II

ComFreek committed:
Fix Travis III

ComFreek committed:
Update bundler version stated in Gemfile.lock

to fix Travis (IV. attempt)

ComFreek committed:
Fix Travis V

ComFreek committed:
Correct spelling of OMDoc

ComFreek committed:
Add Travis badge to readme

ComFreek committed:
Do recommended bundler install at Travis

ComFreek commented:
> Pity about pigments not being supported. We could always consider using Travis and use github-pages as deploy only.

I now added a working Travis setup in my latest commit to this PR: https://github.com/ComFreek/uniformal.github.io/blob/master/.travis.yml

It works on my fork https://comfreek.github.io/uniformal.github.io/doc/ (modulo some path problems with the CSS and JS files - but since this UniFormal repo is hosted "more absolutely" on uniformal.github.io -- and not uniformal.github.io/blah-my-repo-name -- this should *work* here!)

After merging Travis needs to be given a GitHub API key as an env variable under the name `GITHUB_PAGES_DEPLOYMENT_TOKEN`.

**Edit:** I think paths are more broken than expected. Let me investigate before merging.

ComFreek committed:
Fix links once and for all

ComFreek committed:
Adjust url and baseurl for upstream repo [skip ci]

ComFreek commented:
@tkw1536 Finally did it! See https://comfreek.github.io/uniformal.github.io/. Should all be working now!

Remember to set `GITHUB_PAGES_DEPLOYMENT_TOKEN` in Travis and set the pages branch in the repo's settings to `gh-pages` (after the first Travis build succeeded).

ComFreek committed:
Polish readme [skip ci]

ComFreek committed:
Polish readme II [skip ci]
